File: SwrveSDK/SDK/Messaging/SwrveMessageDetails.m
#import "SwrveMessageDetails.h"

@implementation SwrveMessageDetails

@synthesize campaignSubject;
@synthesize campaignId;
@synthesize variantId;
@synthesize messageName;
@synthesize buttons;

- (id)initWith:(NSString *)subject campaignId:(NSUInteger)campId variantId:(NSUInteger)varId messageName:(NSString *)name buttons:(NSMutableArray *)buttonsArray {
    if (self = [super init]) {
        self.campaignSubject = subject;
        self.campaignId = campId;
        self.variantId = varId;
        self.messageName = name;
        self.buttons = buttonsArray;
    }
    return self;
}

@end

File: SwrveSDKSwift/SwrveMessageDetails.swift
@objcMembers public class SwrveMessageDetails: NSObject {
    public var campaignSubject: String
    public var campaignId: UInt
    public var variantId: UInt
    public var messageName: String
    public var buttons: [SwrveMessageButtonDetails]

    public init(with campaignSubject: String, campaignId: UInt, variantId: UInt, messageName: String, buttons: [SwrveMessageButtonDetails]) {
        self.campaignSubject = campaignSubject
        self.campaignId = campaignId
        self.variantId = variantId
        self.messageName = messageName
        self.buttons = buttons
        super.init()
    }
}
